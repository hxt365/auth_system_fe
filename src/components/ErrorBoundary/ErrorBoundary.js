// @flow

import React, { Component } from 'react';
import { Result, Button } from 'antd';

type Props = {
  children: React.ReactNode,
};

class ErrorBoundary extends Component<Props> {
  state = {
    hasError: false,
  };

  componentDidCatch() {
    this.setState({
      hasError: true,
    });
  }

  render() {
    const { hasError } = this.state;
    const { children } = this.props;
    if (hasError)
      return (
        <Result
          status="500"
          title="Crashed!"
          subTitle="Sorry, the server is wrong."
          extra={<Button type="primary">Back Home</Button>}
        />
      );
    return children;
  }
}

export default ErrorBoundary;
