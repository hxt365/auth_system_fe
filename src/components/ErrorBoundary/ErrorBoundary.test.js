import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('test ErrorBoundary', () => {
  it('test snapshot', () => {
    const wrapper = shallow(<ErrorBoundary />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
