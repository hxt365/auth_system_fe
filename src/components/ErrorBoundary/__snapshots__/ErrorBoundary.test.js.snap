// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorBoundary test snapshot 1`] = `""`;
